circus-train-s3-mapreduce-cp/src/main/java/com/hotels/bdp/circustrain/s3mapreducecp/mapreduce/UniformSizeInputFormat.java

```diff
@@ -125,7 +126,7 @@
   private static Path getListingFilePath(Configuration configuration) {
     final String listingFilePathString = configuration.get(S3MapReduceCpConstants.CONF_LABEL_LISTING_FILE_PATH, "");
 
-    assert !listingFilePathString.equals("") : "Couldn't find listing file. Invalid input.";
+    assert !"".equals(listingFilePathString) : "Couldn't find listing file. Invalid input.";
```

Comment:
Remove the assert or throw an illegalArgumentException if this is really bad. I'm fine with removing the line, this is code I think we copied from hadoop so probably ok to keep as is.

Reply:
I agree with removing the `assert`, I guess it all depends on what happens if you do new Path("") and if that's something we really want to avoid (if so then IAE is a good alternative).

Reply:
Im refactoring this to 
```
if ("".equals(listingFilePathString)) {
  throw new IllegalArgumentException("Couldn't find listing file. Invalid input.");
}
```
And I will do similarly for the next ones.